plugins/utils/fzf-lua.nix

```
@@ -0,0 +1,89 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.fzf-lua;
+in {
+  options.plugins.fzf-lua =
+    helpers.extraOptionsOptions
+    // {
+      enable = mkEnableOption "fzf-lua";
+
+      package = helpers.mkPackageOption "fzf-lua" pkgs.vimPlugins.fzf-lua;
+
+      profile = mkOption {
+        type = with types; (enum ["default" "fzf-native" "fzf-tmux" "fzf-vim" "max-perf" "telescope" "skim"]);
+        default = "default";
+        description = "Preconfigured profile to use";
+      };
+
+      keymaps = mkOption {
+        type = with types; attrsOf (either str attrs);
+        description = "Keymaps for Fzf-Lua.";
+        default = {};
+        example = {
+          "<leader>fg" = "live_grep";
+          "<C-p>" = {
+            action = "git_files";
+            actionOpts = {
+              previewers.cat.cmd = "${pkgs.coreutils}/bin/cat";
+              winopts.height = 0.5;
+            };
+            desc = "Fzf-Lua Git Files";
+          };
+        };
+      };
+
+      keymapsSilent = mkOption {
+        type = types.bool;
+        description = "Whether Fzf-Lua keymaps should be silent";
+        default = false;
+      };
+    };
+
+  config = let
+    options =
+      {
+        __unkeyedProfile = cfg.profile;
+      }
+      // cfg.extraOptions;
+  in
+    mkIf cfg.enable {
+      extraPlugins = [cfg.package];
+
+      keymaps =
+        mapAttrsToList
+        (
+          key: action: let
+            actionStr =
+              if isString action
+              then action + "()"
+              else action.action + "(" + helpers.toLuaObject action.actionOpts or {__empty = "";} + ")";
+            actionProps =
+              if isString action
+              then {}
+              else filterAttrs (n: v: n != "action" && n != "actionOpts") action;
+          in {
+            mode = "n";
+            inherit key;
+            action = "function() require('fzf-lua').${actionStr} end";
+            lua = true;
+
+            options =
+              {
+                silent = cfg.keymapsSilent;
+              }
+              // actionProps;
+          }
+        )
+        cfg.keymaps;
```

```suggestion
          key: action: let
            actionStr =
              if isString action
              then "${action}()"
              else "${action.action}(${helpers.toLuaObject action.actionOpts})";
            actionProps =
              if isString action
              then {}
              else filterAttrs (n: v: n != "action" && n != "actionOpts") action;
          in {
            mode = "n";
            inherit key;
            action = "function() require('fzf-lua').${actionStr} end";
            lua = true;

            options =
              {
                silent = cfg.keymapsSilent;
              }
              // actionProps;
          }
        )
        cfg.keymaps;
```

Maybe, we can remove the `keymapsSilent` option as the user can set `actionProps.silent = true`.